### PR TITLE
Feature: make getDefaultLoadFlowProvider return provider from profile when available

### PR DESCRIPTION
**Pull request by jonenst**

### CODE REVIEW

**Review by ayolab — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 1]
```diff
@@ -802,7 +802,19 @@ public boolean setLoadFlowParameters(UUID studyUuid, String parameters, String u
...
+    public String getDefaultLoadflowProvider(String userId) {
+        if (userId != null) {
+            UserProfileInfos userProfileInfos = userAdminService.getUserProfile(userId).orElse(null);
+            if (userProfileInfos != null && userProfileInfos.getLoadFlowParameterId() != null) {
+                try {
+                    return loadflowService.getLoadFlowParameters(userProfileInfos.getLoadFlowParameterId()).getProvider();
+                } catch (Exception e) {
+                    LOGGER.error(String.format("Could not get loadflow parameters with id '%s' from user/profile '%s/%s'. Using default parameters",
+                            userProfileInfos.getLoadFlowParameterId(), userId, userProfileInfos.getName()), e);
+                    // in case of duplication error (ex: wrong/dangling uuid in the profile), move on with default params below
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:814]: remove comment

File: src/main/java/org/gridsuite/study/server/service/StudyService.java [hunk 2]
```diff
@@ -802,7 +802,19 @@ public boolean setLoadFlowParameters(UUID studyUuid, String parameters, String u
...
 
-    public String getDefaultLoadflowProvider() {
+    public String getDefaultLoadflowProvider(String userId) {
+        if (userId != null) {
+            UserProfileInfos userProfileInfos = userAdminService.getUserProfile(userId).orElse(null);
+            if (userProfileInfos != null && userProfileInfos.getLoadFlowParameterId() != null) {
+                try {
+                    return loadflowService.getLoadFlowParameters(userProfileInfos.getLoadFlowParameterId()).getProvider();
+                } catch (Exception e) {
+                    LOGGER.error(String.format("Could not get loadflow parameters with id '%s' from user/profile '%s/%s'. Using default parameters",
```
Comment (dbraquart) [inline src/main/java/org/gridsuite/study/server/service/StudyService.java:812]:
```suggestion
                    LOGGER.error(String.format("Could not get loadflow parameters with id '%s' from user/profile '%s/%s'. Using default provider",
```

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: ok, just a comment to remove
tests: ko, not conform
- "Reset All" button : works as expected
- "Reset provider default values":
  - with no profile : ok
  - with a profile: ko. we still restore all params stored in the profile, while we should set the provider dft values (to be confirmed with the PO ?)

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

**Pull request merged**